### PR TITLE
Cover DSA fields and validation

### PR DESCRIPTION
**Pull request by AntoxaAntoxic**

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/tests/DsaSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        verifyAll {
+            bidderRequests.regs.ext.dsa.required == dsa.required
+            bidderRequests.regs.ext.dsa.datatopub == dsa.datatopub
+            bidderRequests.regs.ext.dsa.pubrender == dsa.pubrender
+            bidderRequests.regs.ext.dsa.transparency[0].domain == dsa.transparency[0].domain
+            bidderRequests.regs.ext.dsa.transparency[0].params == dsa.transparency[0].params
+        }
+
+        where:
+        dsaRequired || bidExt
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/DsaSpec.groovy:52]: Unnessery double vertical bar u can use only one like here `dsaRequired | bidExt`

File: src/test/groovy/org/prebid/server/functional/tests/DsaSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        0           || null
+        1           || new BidExt("dsa": null)
+    }
+
+    def "PBS should populate AMP bid request with DSA regs and send DSA to bidder and always succeed when bidder returns DSA"() {
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/DsaSpec.groovy:57]: 1. PBS amp should return seatbid[].bid[].ext.dsa when DSA is not empty and required is #dsaRequired
2. Please check seat.bid.ext.dsa also

**Review by marki1an — DISMISSED** [review]

**Comment by marki1an** [comment]
Also due to https://github.com/InteractiveAdvertisingBureau/openrtb/blob/main/extensions/community_extensions/dsa_transparency.md specification, few fields can be represent like enum.
Those fields required, pubrender, datatopub, adrender can be enum.

File: src/test/groovy/org/prebid/server/functional/tests/DsaSpec.groovy (hunk 2 above)
Comment (AntoxaAntoxic) [reply src/test/groovy/org/prebid/server/functional/tests/DsaSpec.groovy:57]: As I understand AMP response doesn't contain this info

File: src/test/groovy/org/prebid/server/functional/tests/DsaSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+            actualDsa.behalf == bidDsa.behalf
+            actualDsa.paid == bidDsa.paid
+        }
+
+        and: "PBS should not log warning"
+        assert !response.ext.warnings
+        assert !response.ext.errors
+
+        where:
+        dsaRequired << [ReqsDsaRequiredType.NOT_REQUIRED, ReqsDsaRequiredType.SUPPORTED, ReqsDsaRequiredType.REQUIRED, ReqsDsaRequiredType.REQUIRED_PUBLISHER_ONLINE_PLATFORM]
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/tests/DsaSpec.groovy:120]: can be simplified:
`        dsaRequired << ReqsDsaRequiredType.values()`

File: src/test/groovy/org/prebid/server/functional/tests/DsaSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+        verifyAll {
+            bidderRequests.regs.ext.dsa.required == dsa.required
+            bidderRequests.regs.ext.dsa.dataToPub == dsa.dataToPub
+            bidderRequests.regs.ext.dsa.pubRender == dsa.pubRender
+            bidderRequests.regs.ext.dsa.transparency[0].domain == dsa.transparency[0].domain
+            bidderRequests.regs.ext.dsa.transparency[0].params == dsa.transparency[0].params
+        }
+
+        and: "Bidder response should not contain DSA"
+        def bidderResponse = decode(response.ext.debug.httpcalls.get("generic")[0].responseBody, BidResponse)
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/tests/DsaSpec.groovy:160]: Please don't use hard-coded names. Use instead `BidderName.GENERIC.value`

File: src/test/groovy/org/prebid/server/functional/tests/DsaSpec.groovy [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+        assert bidderResponse.seatbid[0].bid[0].ext?.dsa == null
+
+
```
Comment (osulzhenko) [inline src/test/groovy/org/prebid/server/functional/tests/DsaSpec.groovy:56]: Minor, but there’s a double space. Please use the same approach inside Spec. I suggest changing it to: `assert !bidderResponse.seatbid[0].bid[0].ext?.dsa` Please do the same for other occurrences.

**Review by osulzhenko — CHANGES_REQUESTED** [review]
Pls update code, rest look good

File: src/test/groovy/org/prebid/server/functional/tests/DsaSpec.groovy (hunk 3 above)
Comment (AntoxaAntoxic) [reply src/test/groovy/org/prebid/server/functional/tests/DsaSpec.groovy:120]: I've tried this, but it's not working

Comment (AntoxaAntoxic) [reply src/test/groovy/org/prebid/server/functional/tests/DsaSpec.groovy:120]: never mind, it works)

**Pull request merged**